### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Paul-js-hub**
### In this project I:
- Used decorator design pattern to add new behavior to objects dynamically.
- Implemented composition in the project.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @Paul-js-hub,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### There are no linters running. Please fix your linters setup

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable_interface'
```
Comment (dasileker) [inline decorator.rb:1]: - [ ] Hello and nice job on refactoring the codes but the **require_relative** is not the correct way to call the function please refactor it to `attr_accessor :nameable`

File: person.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
-class Person
+require_relative 'nameable_interface'
```
Comment (dasileker) [inline person.rb:1]: - [ ] Hello and nice job on refactoring the codes but the **require_relative** is not the correct way to call the function please refactor it to `attr_accessor :nameable`

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Paul-js-hub,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Nice PR 👌.
- Linters are green 👌.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @Paul-js-hub 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**